src/index.ts

```diff
@@ -3,8 +3,8 @@
  * @params dom - Dom element.
  * @returns - Html result.
  */
-export default async function domToHtml(dom: Element | null): Promise<string> {
-  if (dom === null || !dom.innerHTML) {
+export default function domToHtml(dom: Element): string {
```

Comment:
Why is this function not async anymore ?

@wadjih-bencheikh18 I just commited a 'test' folder and it seems to me this code is not working.

@targos I would like that this test is available on the project home page and uses a build of the last version of the code but I don't know what is the best way to do this. Any suggestion ?

Reply:
Please also don't use 'default' so that we can easily add other methods

Reply:
What is "the project home page" ?

Reply:
https://github.com/zakodium-oss/dom-to-html

Reply:
It could be a link from this page called 'Demo' for example.

Reply:
I don't understand the point of doing manual testing when we are working on testing with Playwright.

Reply:
> Why is this function not async anymore ?

I removed the async just to test the svg i'm still working on it

Reply:
This library is expected to convert a page containing img, canvas and svg to encode them inline using 'base64' encoded images.

This means that the resulting html should only containing 'img' (so canvas and svg should be replaced) and the content should be base64 encoded.

I don't think that the test are currently checking this.

Also my mistakes that should be fixed:
1. I don't convert <img src=url> to inline img
2. I din't check the <canvas> conversion in the testcase

@wadjih-bencheikh18 could you check those points ?

Reply:
> content should be base64 encoded.

the base64 encoding is not the same for the three devices chromium,firefox and webkit
so i used `expect(resultHtml).toBe(await result.evaluate((node) => node.innerHTML));` so resultHtml (code returned by domToHtml) compared to result content (html result rendred and compared to the test screenshot)

Reply:
after many tests i found that the code is not working with svg tests so i reimplement domToHtml function

Reply:
> I don't think that the test are currently checking this.

i'm checking that it's an `<img/>` and i think snapshots tests are enough to check base64